sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1048,7 +1048,7 @@ class Analyzer(override val catalogManager: CatalogManager)
         }
         // Fail the analysis eagerly because outside AnalysisContext, the unresolved operators
         // inside a view maybe resolved incorrectly.
-        checkAnalysis(newChild)
+        checkAnalysis(newChild, false)
```

Comment:
what's wrong to run the custom check rules? This is kind of a nested query and should go through all the analysis steps.